### ISSUE TITLE
Setter is not called when value is undefined

### ISSUE THREAD

**Issue by lazutkin** [LearnBoost/mongoose#1892]
Setters are not fired with `undefined` value assigned to path.

```coffee
mongoose = require "mongoose"
Schema = mongoose.Schema

Person = new Schema
  name: String

Person.path( "name" ).set ( x ) ->
  console.log "called with value <#{x}> of type #{typeof x}"
  x

Person = mongoose.model "Person", Person


p = new Person( )
p.name = "John"
p.name = undefined
```

Output is:

```
called with value <John> of type string
```

**Comment by JayvicWen** [comment]
+1
It really bothers me a lot.

**Comment by fiznool** [comment]
Is there any chance this fix could be backported to the 3.8 branch?

**Comment by vkarpov15** [comment]
Not likely, this is something that's debateably backwards-breaking and I'm hesitant to put it into 3.8.x as a bug fix.
